parser/v2/templelementparser.go

```diff
@@ -0,0 +1,134 @@
+package parser
+
+import (
+	"fmt"
+	"io"
+	"strings"
+
+	"github.com/a-h/lexical/parse"
+)
+
+var templElementExpressionStartParser = parse.Or(parse.String("<! "), parse.String("<!"))
+var templElementExpressionEndParser = parse.Or(parse.String(" >"), parse.String(">"))
+var templElementExpressionCloseTagParser = parse.Or(parse.String(" />"), parse.String("/>"))
+
+var templBlockElementExpression templBlockElementExpressionParser
+
+type templBlockElementExpressionParser struct{}
+
+func (p templBlockElementExpressionParser) Parse(pi parse.Input) parse.Result {
+	var r TemplElementExpression
+
+	// Check the prefix first.
+	prefixResult := parse.Rune('@')(pi)
+	if !prefixResult.Success {
+		return prefixResult
+	}
+
+	// Once we have a prefix, we must have an expression that returns a template.
```

Comment:
Thinking about this, this is a bit harsh as a parser requirement.

If you have:

```html
<div>
@twitterhandle
</div>
```

This parser will throw an error. I think it would be better if it looked for `@dhjhjkf {` as the start. I'll change it though.